spec/models/admin_spec.rb

```diff
@@ -1,73 +1,88 @@
     end
+  end
 
-    it 'is invalid without a password' do
-      expect(build(:admin, password: nil)).to be_invalid
+  context "length of password" do
+    it "is invalid if under #{ArchiveConfig.PASSWORD_LENGTH_MIN - 1} characters" do
+      expect { create(:admin, password: Faker::Lorem.characters(ArchiveConfig.PASSWORD_LENGTH_MIN - 1)) }.to \
+        raise_error(ActiveRecord::RecordInvalid, \
+                    "Validation failed: Password is too short (minimum is #{ArchiveConfig.PASSWORD_LENGTH_MIN} characters)")
     end
 
-    it 'is invalid without a password confirmation' do
-      expect(build(:admin, password_confirmation: nil)).to be_invalid
+    it "is invalid if over #{ArchiveConfig.PASSWORD_LENGTH_MAX + 1} characters" do
+      expect { create(:admin, password: Faker::Lorem.characters(ArchiveConfig.PASSWORD_LENGTH_MAX + 1)) }.to \
+        raise_error(ActiveRecord::RecordInvalid, \
+                    "Validation failed: Password is too long (maximum is #{ArchiveConfig.PASSWORD_LENGTH_MAX} characters)")
     end
   end
-  #
-  #context "length of login" do
-  #
-  #  let(:login_too_short) {build(:admin, login: Faker::Lorem.characters(2))}
-  #  it "is invalid if under 3 characters" do
-  #    login_too_short.save.should be_false
-  #    login_too_short.errors[:login].should_not be_empty
-  #  end
-  #
-  #  let(:login_too_long) {build(:admin, login: Faker::Lorem.characters(41))}
-  #  it "is invalid if over 40 characters" do
-  #    login_too_long.save.should be_false
-  #    login_too_long.errors[:login].should_not be_empty
-  #  end
-  #end
-  #
-  #context "length of password" do
-  #
-  #  let(:password_too_short) {build(:admin, password: Faker::Lorem.characters(5))}
-  #  it "is invalid if under 6 characters" do
-  #    password_too_short.save.should be_false
-  #    password_too_short.errors[:password].should_not be_empty
-  #  end
-  #
-  #  let(:password_too_long) {build(:admin, password: Faker::Lorem.characters(41))}
-  #  it "is invalid if over 40 characters" do
-  #    password_too_long.save.should be_false
-  #    password_too_long.errors[:password].should be_empty
-  #  end
-  #end
 
   context "uniqueness" do
-    let(:existing_user) {create(:admin)}
+    let(:existing_user) { create(:admin) }
 
     it "is invalid if login is not unique" do
-      expect(build(:admin, login: existing_user.login)).to be_invalid
+      expect { create(:admin, login: existing_user.login) }.to \
+        raise_error(ActiveRecord::RecordInvalid, \
+                    "Validation failed: Login has already been taken")
     end
 
+    # Why does this get a double error ?
     it "is invalid if email already exists" do
-      expect(build(:admin, email: existing_user.email)).to be_invalid
+      expect { create(:admin, email: existing_user.email) }.to \
+        raise_error(ActiveRecord::RecordInvalid, \
+                    "Validation failed: Email has already been taken, Email has already been taken")
```

Comment:
Side-eyeing that repetitive error message. Is that really what it says when you validate one admin?